common/pipeline_storage/src/main/scala/uk/ac/wellcome/pipeline_storage/Indexer.scala

```diff
@@ -25,14 +24,53 @@ trait Indexable[T] {
   def version(document: T): Int
 }
 
+/**
+  * Versioning:
+  *
+  * When the merger makes the decision to merge some works, it modifies
+  * the content of the affected works. Despite the content of these works
+  * being modified, their version remains the same. However, the merger
+  * attaches the number of merged sources (`nMergedSources`) to the target
+  * work and may choose to redirect some of the source works.
+  *
+  * When we ingest those works, we need to make sure that the merger
+  * modified works are never overridden by unmerged works for the same
+  * version still running through the pipeline.
+  *
+  * We also need to make sure that, if a work is modified by a source in
+  * such a way that it shouldn't be merged (or redirected) anymore, the
+  * new unmerged version is ingested and never replaced by the previous
+  * merged version still running through the pipeline.
+  *
+  * We can do that by ingesting works into Elasticsearch with a document
+  * version derived from a combination of work version, number of merged
+  * sources and work type.
+  *
+  * More specifically, by multiplying the work version by 10, we make sure
+  * that a new version of a work always wins over previous versions
+  * (merged or unmerged, provided that there are fewer than 10 merged
```

Comment:
Is this assumption still valid?

Reply:
I _think_ so (I added the bit about < 10 merged versions because that's new) - can you think of a counterexample?

Reply:
Not really, I guess a warning log as suggested by @alexwlchan would be enough for us to spot problems